packages/core/src/api/webhook.ts

```diff
@@ -0,0 +1,167 @@
+			reason,
+			body: webhook,
+		})) as APIWebhook;
+	}
+
+	/**
+	 * Deletes a webhook
+	 *
+	 * @param id - The id of the webhook to delete
+	 * @param token - The token of the webhook
+	 * @param reason - The reason for deleting the webhook
+	 */
+	public async delete(id: string, token?: string, reason?: string) {
+		return this.rest.delete(Routes.webhook(id, token), { reason });
+	}
+
+	/**
+	 * Executes a webhook
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 * @param options - The options to use when executing the webhook
+	 */
+	public async execute(
+		id: string,
+		token: string,
+		options: RESTPostAPIWebhookWithTokenJSONBody & RESTPostAPIWebhookWithTokenQuery & { files?: RawFile[] } = {},
+	) {
+		const { wait, thread_id, files, ...body } = options;
+		return this.rest.post(Routes.webhook(id, token), {
+			query: makeURLSearchParams({ wait, thread_id }),
+			files,
+			body,
+		});
+	}
+
+	/**
+	 * Executes a slack webhook
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 * @param options - The options to use when executing the webhook
+	 */
+	public async executeSlack(id: string, token: string, options: RESTPostAPIWebhookWithTokenSlackQuery = {}) {
+		return this.rest.post(Routes.webhookPlatform(id, token, 'slack'), {
+			query: makeURLSearchParams(options as Record<string, unknown>),
+			body: options,
+		});
+	}
+
+	/**
+	 * Executes a github webhook
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 * @param options - The options to use when executing the webhook
+	 * @returns
+	 */
+	public async executeGitHub(id: string, token: string, options: RESTPostAPIWebhookWithTokenGitHubQuery = {}) {
+		return this.rest.post(Routes.webhookPlatform(id, token, 'github'), {
+			query: makeURLSearchParams(options as Record<string, unknown>),
+			body: options,
+		});
+	}
+
+	/**
+	 * Fetches an associated message from a webhook
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 * @param messageId - The id of the message to fetch
+	 * @param options - The options to use when fetching the message
+	 */
+	public async getMessage(id: string, token: string, messageId: string, options: { thread_id?: string } = {}) {
+		return (await this.rest.get(Routes.webhookMessage(id, token, messageId), {
+			query: makeURLSearchParams(options as Record<string, unknown>),
+		})) as APIMessage;
+	}
+
+	/**
+	 * Edits an associated message from a webhook
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 * @param messageId - The id of the message to edit
+	 * @param options - The options to use when editing the message
+	 */
+	public async editMessage(
+		id: string,
+		token: string,
+		messageId: string,
+		options: RESTPatchAPIWebhookWithTokenMessageJSONBody & { thread_id?: string } = {},
+	) {
+		const { thread_id, ...body } = options;
```

Comment:
Should this actually be optional?
```suggestion
		{ thread_id, ...body }: RESTPatchAPIWebhookWithTokenMessageJSONBody & { thread_id?: string },
	) {
```